### ISSUE TITLE
GPU Ram exhaustion TF 2.1 CUDA 10.1

### ISSUE THREAD

**Issue by aconley** [tensorflow/tensorflow#36602]
**System information**
- Have I written custom code : yes, although very minimal
- OS Platform and Distribution: Linux Ubuntu 19.10
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.1.0  (also tested 2.0.0)
- Python version: 3.6.10 anaconda
- CUDA/cuDNN version: 10.1.243
- GPU model and memory: Nvidia RTX 2080 7979MiB

**Describe the current behavior**
Since updating to 2.1.0, the following simple commands cause memory exhaustion and
all further inputs to fail with cuDNN errors:

```python
X = tf.random.normal((16, 128, 16), dtype=tf.float32)
layer_norm = tf.keras.layers.LayerNormalization()
Xn = layer_norm(X)
```
which causes
cuDNN launch failure : input shape ([1,2048,16,1]) [Op:FusedBatchNormV3]
(BatchNormalization has the same issue, unsurprisingly)

Initially, I thought this was a CUDA/CuDNN installation issue, but if I add

```python
for gpu in tf.config.experimental.list_physical_devices('GPU'):
    tf.config.experimental.set_memory_growth(gpu, False)
```

the problem does not occur.

This issue does not happen if I change TF or CUDA versions TF 2.0.0 or CUDA 10.0 
-- it is specific to 2.1.0 + 10.1.  (It is still present in the current nightly).

I have not been able to replicate this problem in, say, colab, but I don't know what version
of Cuda is in use there.

Test cases:
  * TF 2.1.0 Cuda 10.1.243 CuDNN(-Dev) 7.6.5.42-1+cuda10.1 TensorRT 6.0.1.5+cuda10.1
     * Memory Growth: True ->  Failure with CuDNN start error
     * Memory Growth: False -> Success
  * TF 2.0.0 Cuda 10.1.243 CuDNN(-Dev) 7.6.5.42-1+cuda10.1 TensorRT 6.0.1.5+cuda10.1
     * Memory Growth: True ->  Success
     * Memory Growth: False -> Success
  * TF 2.1.0 Cuda 10.0.130 CuDNN(-Dev) 7.6.5.32-1+cuda10.0 TensorRT 7.0.0.11
     * Memory Growth: True ->  Success
     * Memory Growth: False -> Success
  *  TF 2.0.0 Cuda 10.0.130 CuDNN(-Dev) 7.6.5.32-1+cuda10.0 TensorRT 7.0.0.11
     * Memory Growth: True ->  Success
     * Memory Growth: False -> Success
  *  TF 2.0.0 Cuda 10.0.130 CuDNN(-Dev) 7.6.5.32-1+cuda10.0 TensorRT 7.0.0.11
     * Memory Growth: True ->  Success
     * Memory Growth: False -> Success
  *  TF 2.1.0 Cuda 10.0.130 CuDNN(-Dev) 7.6.5.32-1+cuda10.0 TensorRT 6.0.1.5
     * Memory Growth: True ->  Success
     * Memory Growth: False -> Success

Note TensorRT 7 is not available for CUDA 10.1, hence the last test with v6 of TensorRT.
I tried to test on CUDA 10.2 by building from source, but got bogged down with blaze
installation conflicts.  I have reinstalled the failing set of libraries + TF from scratch more than
once as a sanity check, and it has re-occurred each time.

**Describe the expected behavior**
  It should be possible to execute LayerNorm/BatchNorm.

**Comment by aconley** [comment]
Sorry, I flipped True and False for memory growth in the above.  Aagh.

That should be:

```python
for gpu in tf.config.experimental.list_physical_devices('GPU'):
    tf.config.experimental.set_memory_growth(gpu, True)
```
fixes the problem for 2.1.0/10.1, and

* TF 2.1.0 Cuda 10.1.243 CuDNN(-Dev) 7.6.5.42-1+cuda10.1 TensorRT 6.0.1.5+cuda10.1
    * Memory Growth: False -> Failure with CuDNN start error
    * Memory Growth: True -> Success

**Comment by ymodak** [comment]
Yes. Limiting gpu growth by setting flag as `true` is correct approach. Closing this issue since its resolved. Thanks!

**Comment by aconley** [comment]
Verified that the TF 2.1 with CUDA 10.0 was only passing because it was using the CPU.

However: also verified that TF 2.0 + CUDA 10.0 does work without needing to limit GPU growth.  So needing to limit memory *is* new behavior in 2.1 + CUDA 10.1.
